### PR TITLE
add patch for PyQtGraph to fix exporting images

### PR DESCRIPTION
**Pull request by SimonPinches**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/p/PyQtGraph/PyQtGraph-0.10.0-ImageExporter.patch [hunk 1]
```diff
@@ -0,0 +1,12 @@
+diff -Nru pyqtgraph-0.10.0-orig/pyqtgraph/exporters/ImageExporter.py pyqtgraph-0.10.0/pyqtgraph/exporters/ImageExporter.py
```
Comment (lexming) [inline easybuild/easyconfigs/p/PyQtGraph/PyQtGraph-0.10.0-ImageExporter.patch:1]: Please add authorship, description and link to bug report (if any) for this patch.

**Review by lexming — CHANGES_REQUESTED** [review]
Thanks for submitting this patch. Would it be possible to add it to all easyconfigs of PyQtGraph and not only to `2018a`?

The tests fail because there have been some changes in our requirements for easyconfigs. Now all `PythonPackage` should have `sanity_pip_check = True`.
Moreover, the current `sanity_check_paths` of this easyconfig are now default, so those could be removed.

File: easybuild/easyconfigs/p/PyQtGraph/PyQtGraph-0.10.0-ImageExporter.patch (hunk 1 above)
Comment (SimonPinches) [reply easybuild/easyconfigs/p/PyQtGraph/PyQtGraph-0.10.0-ImageExporter.patch:3]: @lexming I've just pushed the requested changes.  Let me know if more are needed.

File: easybuild/easyconfigs/p/PyQtGraph/PyQtGraph-0.10.0-ImageExporter.patch [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
++++ pyqtgraph-0.10.0/pyqtgraph/exporters/ImageExporter.py	2020-06-24 10:50:36.544574000 +0200
+@@ -67,7 +67,7 @@
+         w, h = self.params['width'], self.params['height']
+         if w == 0 or h == 0:
+             raise Exception("Cannot export image with size=0 (requested export size is %dx%d)" % (w,h))
+-        bg = np.empty((self.params['width'], self.params['height'], 4), dtype=np.ubyte)
++        bg = np.empty((int(self.params['width']), int(self.params['height']), 4), dtype=np.ubyte)
+         color = self.params['background']
+         bg[:,:,0] = color.blue()
+         bg[:,:,1] = color.green()
```
Comment (lexming) [inline easybuild/easyconfigs/p/PyQtGraph/PyQtGraph-0.10.0-ImageExporter.patch:14]:
```suggestion
@@ -64,10 +64,11 @@
         
         #self.png = QtGui.QImage(targetRect.size(), QtGui.QImage.Format_ARGB32)
         #self.png.fill(pyqtgraph.mkColor(self.params['background']))
-        w, h = self.params['width'], self.params['height']
+        w = int(self.params['width'])
+        h = int(self.params['height'])
         if w == 0 or h == 0:
             raise Exception("Cannot export image with size=0 (requested export size is %dx%d)" % (w,h))
-        bg = np.empty((self.params['width'], self.params['height'], 4), dtype=np.ubyte)
+        bg = np.empty((h, w, 4), dtype=np.ubyte)
         color = self.params['background']
         bg[:,:,0] = color.blue()
         bg[:,:,1] = color.green()
```

File: easybuild/easyconfigs/p/PyQtGraph/PyQtGraph-0.10.0-ImageExporter.patch [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# Patch to fix exporting images
+# D.Penko, S.D.Pinches, 2019
```
Comment (lexming) [inline easybuild/easyconfigs/p/PyQtGraph/PyQtGraph-0.10.0-ImageExporter.patch:2]:
```suggestion
# Ensure integer size of exported images. See https://github.com/pyqtgraph/pyqtgraph/pull/356
# authors: D.Penko, S.D.Pinches, 2019
```

**Review by lexming — CHANGES_REQUESTED** [review]
Thanks for updating the PR. I revised the patch and I think that it is safer to use the same approach found in PyQtGraph v0.11.0, as it converts to `int` before the conditional check. Otherwise values between 0 and 1 could cause issues.

**Comment by boegel** [comment]
Just as a reminder: updating the patch file implies updating the checksum for the patch file as well.

@lexming With that in mind, maybe it's easier to send a proper PR to @SimonPinches' branch used for this PR, rather than using the suggestions feature?

**Comment by lexming** [comment]
@boegel you are right, I totally forgot about the checksum. I'll make a PR

**Comment by lexming** [comment]
@SimonPinches I made a PR into the branch of this PR with my suggestions, so you can easily incorporate them. Please check https://github.com/SimonPinches/easybuild-easyconfigs/pull/14

**Comment by SimonPinches** [comment]
Thanks @lexming.

**Comment by lexming** [comment]
Test report by @lexming
**FAILED**
Build succeeded for 6 out of 9 (4 easyconfigs in this PR)
node379.hydra.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/593cf55fb40f420c76cd755c1e0838f3 for a full test report.

**Comment by lexming** [comment]
Test report by @lexming
**FAILED**
Build succeeded for 3 out of 5 (4 easyconfigs in this PR)
node127.hydra.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2680 v2 @ 2.80GHz, Python 2.7.5
See https://gist.github.com/562c2dfea895d66cd90f284802b3193c for a full test report.

**Comment by lexming** [comment]
Test report by @lexming
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
node128.hydra.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2680 v2 @ 2.80GHz, Python 2.7.5
See https://gist.github.com/f1ff38085511b2ca714a2d71d248e093 for a full test report.

**Review by lexming — APPROVED** [review]
LGTM

**Comment by lexming** [comment]
Going in, thanks @SimonPinches !

**Pull request merged**